frontend/assets/components/piiDetect.js

```diff
@@ -0,0 +1,32 @@
+import { getCommentTexts } from "../apis/zafApis.js";
+import { detectPiiEntities as comprehendModel } from "../apis/comprehendApis.js";
+import { stripHtml, concatStrings } from "../utils.js";
+import {
+  commentsConcatDelimiter,
+  entitiesConcatDelimiter,
+} from "../constants.js";
+
+const textPreprocessor = (texts) => {
+  const cleanTexts = texts.map((text) => stripHtml(text));
+  return concatStrings(cleanTexts, commentsConcatDelimiter);
+};
+
+const predictPiiEntities = async (text, modelName) => {
+  // use return to show error to users
+  if (modelName === "Spacy") return new Error("Spacy model is not implemented");
```

Comment:
would be good to have one spot to define the different modelNames as variables, instead of relying on string mathcing between here and the dropdown.

Reply:
I've created `dropdown.js` in which contains all of the supported models, details at a20c0580fa4e9380a217c9a5186837177726c76a, is this implementation you have in mind as well?